provider/gateway/client.go

```diff
@@ -282,3 +414,12 @@ func (c *client) ServiceLogs(ctx context.Context,
 
 	return logs, nil
 }
+
+func parseCloseMessage(msg string) string {
+	errmsg := strings.SplitN(msg, ": ", 3)
```

Comment:
It'd be helpful to have a comment showing an example of what is being parsed out here